python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -18,27 +18,26 @@
 from contextlib import contextmanager
 import functools
 
-import Ganga.Utility.logging
+from Ganga.Utility.logging import getLogger
 
 from copy import deepcopy, copy
-import inspect
+from inspect import isclass
 
-import Ganga.GPIDev.Schema as Schema
+from Ganga.GPIDev.Schema import Schema, Item, ComponentItem, SharedItem
 
 from Ganga.Core.exceptions import GangaValueError, GangaException
 
 from Ganga.Utility.Plugin import allPlugins
 
 def _getName(obj):
     """ Return the name of an object based on what we prioritise"""
-    returnable = getattr(obj, '_name', getattr(obj, '__name__', None))
+    returnable = getattr(obj, '_name', getattr(obj, '__name__', getattr(getattr(obj, '__class__', None), '__name__', None)))
```

Comment:
This was used to replace situations where the object can have either. I can drop this back to multiple line statements if you're not a fan of this.